lib/bacalhau.js

```diff
@@ -0,0 +1,162 @@
+import timers from 'node:timers/promises'
+import { execa } from 'execa'
+import { fetch } from 'undici'
+import * as Sentry from '@sentry/node'
+import { installBinaryModule, getBinaryModuleExecutable } from './modules.js'
+
+const DIST_TAG = 'v0.3.28'
+
+export async function install () {
+  await installBinaryModule({
+    module: 'bacalhau',
+    repo: 'bacalhau-project/bacalhau',
+    executable: 'bacalhau',
+    distTag: DIST_TAG,
+    targets: [
+      { platform: 'darwin', arch: 'x64', asset: `bacalhau_${DIST_TAG}_darwin_amd64.tar.gz` },
+      { platform: 'darwin', arch: 'arm64', asset: `bacalhau_${DIST_TAG}_darwin_arm64.tar.gz` },
```

Comment:
If they arm64 binary is not signed & notarised, then I won't be able to run it on my machine. We may need to use the same platform override as we do for Saturn L2.

I'll run this PR locally on my machine to see if it is a problem or not.

We can land this PR as-is; adding such an override can be easily done later.

Reply:
It did work for me though, and I'm on a M2

Reply:
This depends on your system's security settings too. 

- https://support.apple.com/en-us/HT202491
- https://gadgetstouse.com/blog/2021/02/05/run-unverified-unidentified-developer-apps-on-mac/#3_terminal_command